playbooks/common/openshift-master/config.yml

```diff
@@ -122,6 +122,7 @@
       openshift_master_certs_no_etcd:
       - admin.crt
       - master.kubelet-client.crt
+      - master.proxy-client.crt
```

Comment:
this template should do the trick, of course I'm making up a variable containing the OpenShift version (not sure if one already exists):
- "{{ 'master.proxy.client.crt' if openshift_version >= 3.1 else omit }}"

The omit keyword should cause that loop iteration to be a noop when the version is less than 3.1. of course, using the version comparison filter is probably the right way to compare as well.